pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -119,6 +120,23 @@ func shutdownAppsGlobal(ctx *zedagentContext) {
 	shutdownApps(ctx.getconfigCtx)
 }
 
+func parseBaseOS(getconfigCtx *getconfigContext,
+	config *zconfig.EdgeDevConfig) {
+
+	baseOS := config.GetBaseos()
+	var newRetryUpdateCounter uint32
+	if baseOS != nil && baseOS.GetRetryUpdate() != nil {
```

Comment:
If nil shouldn't you return? Assuming the counter == 0 seems wrong.

Reply:
But what we should assume in case of nil? Keep old values?

Reply:
Yes, because you didn't receive anything.

Reply:
But shouldn't I unpublish novel baseOs from pubsub in this case? Or keep the old one?

Reply:
Just log and return.

Reply:
Ok, rework it this way